src/daft-plan/src/treenode.rs

```diff
@@ -2,7 +2,67 @@ use std::sync::Arc;
 
 use crate::{physical_plan::PhysicalPlan, LogicalPlan};
 use common_error::DaftResult;
-use common_treenode::DynTreeNode;
+use common_treenode::{DynTreeNode, Transformed, TreeNode, TreeNodeIterator};
+
+impl TreeNode for LogicalPlan {
```

Comment:
we already implement `DynTreeNode` and therefore `TreeNode` for `LogicalPlan` and `PhysicalPlan`. We shouldn't need to have to implement the trait twice?

Reply:
we don't actually have `TreeNode` for `LogicalPlan` but instead `Arc<LogicalPlan>`, so this causes an issue where i want to blanket impl the visitor for all types that are `TreeDisplay + TreeNode`

```rs
impl<T> TreeNodeVisitor for MermaidDisplayVisitor<T>
where
    T: TreeDisplay + TreeNode,
{}
```

but `TreeDisplay` is implemented for `LogicalPlan` and `TreeNode` for `Arc<LogicalPlan>`, so one of them had to change. Some of the other changes in the PR are a result of implementing `TreeNode` for `LogicalPlan`

Reply:
I see. What do you think about making TreeDisplay always being a TreeNode.
```
pub trait TreeDisplay: TreeNode 
```
And then rather having `MermaidDisplayVisitor` being a generic, we can make take in `&dyn TreeDisplay` for the methods. This will make it work regardless if it's actually a `DynTreeNode` or `TreeNode`

Reply:
but then that just pushes the issue of `TreeNode` not being implemented for `LogicalPlan` elsewhere?